yarp/yarp_compiler.c

```diff
@@ -12,6 +12,15 @@
 #define NEW_CHILD_ISEQ(node, name, type, line_no) \
     yp_new_child_iseq(iseq, (node), parser, rb_fstring(name), iseq, (type), (line_no))
 
+#define YP_COMPILE(node) \
```

Comment:
Very minor, but you might want to consider:

```suggestion
#define YP_COMPILE_BASE(node, popped) \
    yp_compile_node(iseq, (node), ret, src, popped, compile_context)

#define YP_COMPILE(node) YP_COMPILE_BASE(node, popped)
#define YP_COMPILE_POPPED(node) YP_COMPILE_BASE(node, true)
#define YP_COMPILE_NOT_POPPED(node) YP_COMPILE_BASE(node, false)
```